flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraph.java

```diff
@@ -352,8 +353,8 @@ public JobVertex findVertexByID(JobVertexID id) {
 	 *
 	 * @param paths paths of the directories/JAR files required to run the job on a task manager
 	 */
-	public void setClasspaths(List<URL> paths) {
-		classpaths = paths;
+	public void addClasspaths(Collection<URL> paths) {
```

Comment:
Are we actually invoking `addClasspaths` multiple times? It seems that the changeset could be smaller by just adding a null check to the setter.

Reply:
So for `addJars`. We don't call `addJars` multiple times for now. However, it fits the abstraction that we firstly instance an empty collection and then add member per call. If we still using setter and with a null checker the field cannot be `final` while we don't gain a lot from such mutability.